cmd/util/ledger/migrations/add_key_migration.go

```diff
@@ -0,0 +1,391 @@
+package migrations
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/cadence/common"
+	"github.com/onflow/crypto/hash"
+
+	"github.com/onflow/flow-go-sdk/crypto"
+	"github.com/onflow/flow-go/cmd/util/ledger/reporters"
+	"github.com/onflow/flow-go/cmd/util/ledger/util/registers"
+	"github.com/onflow/flow-go/fvm"
+	"github.com/onflow/flow-go/fvm/systemcontracts"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// This migration is not safe to run on actual networks.
+// It is used for getting control over system accounts so that they can be tested
+// in a copied environment. When we need to do this it is best to create a branch
+// where this variable is set to true, and use that temporary branc to run migrations.
+const IAmSureIWantToRunThisMigration = false
+
+// AddKeyMigration adds a new key to the core contracts accounts
+type AddKeyMigration struct {
+	log zerolog.Logger
+
+	accountsToAddKeyTo map[common.Address]AddKeyMigrationAccountPublicKeyData
+	chainID            flow.ChainID
+
+	reporter reporters.ReportWriter
+}
+
+var _ AccountBasedMigration = &AddKeyMigration{}
+
+func NewAddKeyMigration(
+	chainID flow.ChainID,
+	key crypto.PublicKey,
+	rwf reporters.ReportWriterFactory,
+) *AddKeyMigration {
+	if !IAmSureIWantToRunThisMigration {
+		panic("Cannot run AddKeyMigration migration")
+	}
+
```

Comment:
Better to add another "checkbox" to remind the required step:
```suggestion
	if !IHaveReplacedEmptyNodeAddressesToMainnetAddresses {
		panic("didn't use the mainnet node addreses")
	}
```